src/compiler/checker.ts

```diff
@@ -49476,7 +49476,6 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                         return true;
                     }
                 }
-                return false;

```

Comment:
That's the core of the fix for what was reported. `for await` loops never had a chance to reach `checkGrammarVariableDeclarationList` call in this function. Perhaps there are some other subtle issues in this function? Shouldn't the rule of thumb be here only to use `return true` as an early return?